src/app/reporting/SynchronizedReportSchedulerImpl.h

```diff
@@ -30,6 +30,48 @@ using Milliseconds64  = System::Clock::Milliseconds64;
 using ReadHandlerNode = ReportScheduler::ReadHandlerNode;
 using TimerDelegate   = ReportScheduler::TimerDelegate;
 
+/**
+ * @class Synchronized ReportSchedulerImpl
+ *
+ * @brief This class extends ReportSchedulerImpl and overrides it's scheduling logic.
+ *
+ * It only overrides Observers method where the scheduling logic make it necessary, the others are kept as is.
+ *
+ * It inherits from TimerContext so that it can be used as a TimerDelegate instead on relying on the nodes to schedule themselves.
+ *
+ * ## Scheduling Logic
+ *
+ * This class implements a scheduling logic that aims to make all ReadHandlers report at the same time when possible.
+ * The goal is to minimize the different times a device wakes up to report, and thus this aims to schedule all reports at the latest
+ * possible time while ensuring that all reports get sent before their max interval.
+ *
+ * The logic also aims to minimize the impact on the responsivity of the device.
+ *
+ * The scheduling logic is as follows:
+ * - The CalculateNextReportTimeout is called by the same ReadHandler Observer callbacks than the non-synchronized implementation:
+ *      * OnSubscriptionEstablished,
+ *      * OnBecameReportable,
+ *      * OnSubscriptionReportSent
+ *
+ * - The Synchronized Scheduler keeps track of the next min and max interval timestamps. It updates in CalculateNextReportTimeout
+ *
+ * - The next max interval is calculated as the earliest max interval of all the registered ReadHandlersNodes.
+ *
+ * - The next min interval is calculated as the latest min interval of the registered ReadHandlersNodes that:
+ *     * Have a min timestamp greater than the current time
+ *     * Are Reportable (this prevents a ReadHandler that is not reportable to hold the report of all the others)
```

Comment:
The "to hold" part is not making sense to me.  Should that be "prevents a ReadHandler that is not reportable from blocking reporting of other ReadHandlers"?

Reply:
Yes, preventing is clearer.